soccer/gameplay/evaluation/passing_positioning.py

```diff
@@ -118,57 +80,25 @@ def eval_singl_point(kick_point, receive_point, ignore_robots, field_weights,
 ## Finds the best position to pass to
 #
 # @param kick_point: Point that we are passing from
-# @param evaluation_zone: Area to evaluate passes to
 # @param ignore_robots: Robots to ignore when calculating scores
 # @param field_weights: A tuple of the 3 difference weights to apply to field position 
 #               (Centerness, Distance to their goal, Angle off their goal)
 # @param weights: A tuple of the 4 different weights to apply to the evaulations overall (Weights are normalized)
 #               (space, field_position, shot_chance, kick_proximity)
-# @param debug: Displays pass lines and scores onto the field when true
 # @return bestPoint and bestScore in that order
 def eval_best_receive_point(kick_point,
-                            evaluation_zone=None,
                             ignore_robots=[],
                             field_weights=(0.1, 3.2, 0.1),
-                            weights=(1, 4, 15, 1),
-                            debug=False):
-
-    if evaluation_zone is None:
-        evaluation_zone = generate_default_rectangle(kick_point)
-
-    points = get_points_from_rect(evaluation_zone, kick_point, 0.6, 0.5)
-
-    if points is None or len(points) == 0:
-        # Nothing can be done
-        return None, 0
-
-    # TODO: Setup to be a list of the top X points
-    bestScore = None
-    bestPoint = None
-
-    # Finds best score out of all the points
-    for currentPoint in points:
-        currentScore = eval_singl_point(kick_point, currentPoint,
-                                        ignore_robots, field_weights, weights)
-
-        if (debug):
-            score_color = (round(currentScore * 255), 0,
-                           round((1 - currentScore) * 255))
-            main.system_state().draw_line(
-                robocup.Segment(kick_point, currentPoint), score_color,
-                "Score")
-
-        if bestScore is None or currentScore > bestScore:
-            bestScore = currentScore
-            bestPoint = currentPoint
-
-    # Returns None if we can't find anything
-    if bestPoint is None:
-        return None, 0
-
-    if (debug):
-        main.system_state().draw_line(
-            robocup.Segment(kick_point, bestPoint), constants.Colors.Red,
-            "Best Point")
-
-    return bestPoint, bestScore
+                            weights=(1, 4, 15, 1)):
+    pythfunc = functools.partial(eval_single_point, kick_point, ignore_robots, \
+                field_weights, weights)
+    cppfunc = robocup.stdfunction(pythfunc)
+    nmConfig = robocup.NelderMead2DConfig(cppfunc, \
+                                        kick_point, \
+                                        robocup.Point(0.5, 2), \
+                                        robocup.Point(0.01, 0.01), \
+                                        1, 2, 0.75, 0.5, 50, 1, 0.1)
```

Comment:
I should probably change these to arguments of `eval_best_recieve_point` because they are dependent on the function being optimized. The biggest change that I did was messing with the starting points of the triangle. Most of the other arguments are the default values I took from wikipedia.

Nelder Mead can be visualized as a triangle being flipped and resized as it moves around trying to find the best point. [Here](https://en.wikipedia.org/wiki/Nelder%E2%80%93Mead_method#/media/File:Nelder-Mead_Simionescu.gif) is a decent visualization of it.

As for the arguments themselves...
`kick_point` is the starting point of the triangle. 
`Point(0.5, 2)` describes where the other two points are. I want to say the X axis is flipped so the two points are +-0.5 , 2 in reference to the staring point. 
`Point(0.01, 0.01)` is the exit condition. Once the bounding box of the triangle is this size, the algorithm exits. 
`1` is the scalar value for a reflection of the triangle.
`2` is the scalar value for an expansion of a point in the triangle. 
`.75` is the contraction scalar value for a point in the triangle. 
`.5` is the shrink value for all points in the triangle. 
`50` is the max number of iterations. 
`1` and `0.1` specify the max value of the function and a range so we can exit the optimization early if we are close to the max.

To test these, I've been running `test_adaptive_formation_weights` and just checking where the white square is in comparison to where I think it should be.